src/DotVVM.Framework/Controls/RouteLink.cs

```diff
@@ -119,5 +119,17 @@ protected override void RenderContents(IHtmlWriter writer, IDotvvmRequestContext
                 }
             }
         }
+
+        protected virtual void WriteOnClickAttribute(IHtmlWriter writer, IDotvvmRequestContext context)
+        {
+            if ((bool)GetValue(Internal.IsSpaPageProperty) && context.Configuration.UseHistoryApiSpaNavigation)
+            {
+                writer.AddAttribute("onclick", "return !this.hasAttribute('disabled') && dotvvm.handleSpaNavigation(this.getAttribute('href'));");
+            }
+            else
+            {
+                writer.AddAttribute("onclick", "return !this.hasAttribute('disabled');");
+            }
```

Comment:
I think that it should be enough to rely on the `popstate` event.

Reply:
It is necessary to distinguish links inside and outside of the SPA

Reply:
1. If you are handling popstate event correctly, it should invoke the SPA navigation anyway.
2. You are rendering this handler for any route, regardless of it's target page.

Reply:
>If you are handling popstate event correctly, it should invoke the SPA navigation anyway.

As much as I know, `popstate` event is not invoked on navigation via link

>You are rendering this handler for any route, regardless of it's target page.

You're right